### PR TITLE
Feature/unit tests

### PR DESCRIPTION
**Pull request by petevg**
No longer a Work In Progress; this is ready for review!

These are basic tests for layer-apache-bigtop-base. I'd appreciate feedback on technique (too mock-y, just mock-y enough?).

@juju-solutions/bigdata

### CODE REVIEW

File: lib/charms/layer/apache_bigtop_base.py [hunk 1]
```diff
@@ -19,12 +19,22 @@ class Bigtop(object):
...
+    @property
+    def bigtop_base(self):
+        '''Path to the bigtop root directory.'''
+        return self._bigtop_base
+
+    @property
+    def site_yaml(self):
+        '''Path to the site_yaml config file.'''
+        return self._site_yaml
+
```
Comment (petevg) [inline lib/charms/layer/apache_bigtop_base.py]: Moved these here to make them override-able in the tests. Also helps with introspection.

Comment (petevg) [inline lib/charms/layer/apache_bigtop_base.py]: (The self.site_yaml stuff will actually conflict when https://github.com/juju-solutions/layer-apache-bigtop-base/pull/24 gets merged -- will update this when that happens.)

**Comment by kwmonroe** [comment]
This looks great for the bigtop base layer, but we have a problem for any charm built from it.  The tox.ini here is overwritten by the tox.ini in layer-basic, which clobbers stuff like the `-runit_test_requiresments.txt` dep handler.

This is a problem because `tox` on a built charm will fail when it tries to run `py.test -v` and hits the `./tests/unit/*` stuff brought in by the bigtop base layer.  I suggest adding ./tests/unit to the `ignores` for the bigtop-base layer so these tests don't wind up in the built charms.

```
--- a/layer.yaml
+++ b/layer.yaml
@@ -2,6 +2,8 @@ includes:
   - 'layer:basic'
   - 'layer:puppet-agent'
   - 'interface:java'
+ignore:
+  - 'tests/unit'
 options:
   basic:
     packages:
```

File: README.md [hunk 2]
```diff
@@ -84,9 +84,9 @@ As of this writing, this layer does not define any actions.
 
 ## Unit Tests
 
-To run unit tests for this layer, simple change to the root directory
-of the layer in a terminal, and run "tox". To tweak settings, such as
-making the tests more or less verbose, edit tox.ini.
+To run unit tests for this layer, change to the root directory of the
+layer in a terminal, and run "tox -c tox_unit.ini". To tweak settings,
+such as making the tests more or less verbose, edit tox-unit.ini.
```
Comment (kwmonroe) [inline README.md]: `tox-unit.ini`?  file not found.

:)

**Comment by kwmonroe** [comment]
LGTM

**Pull request merged**